api/ifaces/listeners.go

```diff
@@ -7,8 +7,28 @@ import (
 )
 
 type AppListener interface {
```

Comment:
These listeners are now dealt individually. Is this being used anywhere else? If not, what do you say about removing it?